### PR TITLE
chore(deps): migrate to nx 17

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Keep nx up to date

## This pull request

- Run nx recommended commands to upgrade nx to latest version. `nx migrate nx@17.2.3`

## Issue that this pull request solves

Closes: # (issue number)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: nx.json [hunk 1]
```diff
@@ -37,17 +17,20 @@
         "{projectRoot}/src/styles/*.css",
         "{projectRoot}/test/**/*.ftl",
         "{projectRoot}/test/**/client.json"
-      ]
+      ],
+      "cache": true
```
[the remark this answers is not in the document]
Comment (StaberindeZA) [reply nx.json:21]: Does appear so. I had a quick look at release notes and Nx docs, and doesn't look like `NX_SKIP_NX_CACHE` behavior has changed.

From my understanding, this just moves the config from the "taskRunnerOptions" property, to the individual targets.

File: package.json [hunk 2]
```diff
@@ -170,10 +170,10 @@
     "@types/babel__core": "^7",
     "@types/jest": "^29.5.1",
     "@types/mysql": "^2",
-    "@types/node": "^18.18.2",
+    "@types/node": "18.16.9",
```
Comment (chenba) [inline package.json:173]: Why the downgrade?

Comment (StaberindeZA) [reply package.json]: Ah, must have happened as part of the nx migration. I've reverted the change.

**Review by chenba — APPROVED** [review]
Thanks for the upgrade.

**Comment by StaberindeZA** [comment]
Had some issues with caches from the build step, not being picked up by follow on jobs, causing the Functional Tests CI job to time out.

Closing this in favor of attempting migration to Nx 18 which released a few days ago.

**Pull request closed without merging**